### PR TITLE
feat: make the gRPC channel between Frontend and Datanode configurable

### LINKED ISSUE #1717: Make the gRPC channel between Frontend and Datanode configurable
### What type of enhancement is this?

Tech debt reduction

### What does the enhancement do?

Currently the values for `ChannelConfig` are hard coded in `DatanodeClients`, better made them configurable in Frontend's config file.

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by sunray-ley**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This PR exposes `DatanodeOptions` from the Frontend, adds configuration options to the configuration file. As a result, the gRPC channel between Frontend and Datanode becomes configurable.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

Closes #1717

### CODE REVIEW

**Review by killme2008 — APPROVED** [review]
LGTM

**Review by MichaelScofield — APPROVED** [review]

File: src/frontend/src/service_config/datanode.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default)]
+pub struct DatanodeOptions {
+    client_options: DatanodeClientOptions,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DatanodeClientOptions {
+    pub timeout_millis: u64,
+    pub connect_timeout_millis: u64,
```
Comment (evenyag) [inline src/frontend/src/service_config/datanode.rs:26]: suggestion: Could we use `humantime_serde` to take a readable duration?

https://github.com/GreptimeTeam/greptimedb/blob/81ea61ba43f5f62bb509cdbb58e1b779c1b5a1f4/src/servers/src/http.rs#L135-L139

File: src/frontend/src/service_config/datanode.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_grpc::channel_manager;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Default)]
+pub struct DatanodeOptions {
+    client_options: DatanodeClientOptions,
```
Comment (evenyag) [inline src/frontend/src/service_config/datanode.rs:20]: style(non-blocking): I think the `options` suffix is unnecessary.
```suggestion
    client: DatanodeClientOptions,
```

**Comment by MichaelScofield** [comment]
@sunray-ley could you add these options to our documents as well?
https://docs.greptime.com/user-guide/operations/configuration#frontend-in-distributed-mode

related issue: https://github.com/GreptimeTeam/docs/issues/504

**Comment by sunray-ley** [comment]
> @sunray-ley could you add these options to our documents as well? https://docs.greptime.com/user-guide/operations/configuration#frontend-in-distributed-mode
> 
> related issue: [GreptimeTeam/docs#504](https://github.com/GreptimeTeam/docs/issues/504)

Sure, I'll be happy to update the documentation with the required options.

File: src/frontend/src/service_config/datanode.rs (hunk 1 above)
Comment (sunray-ley) [reply src/frontend/src/service_config/datanode.rs]: Thanks for the suggestion! I've used `humantime_serde` for readable durations. The PR has been updated with the changes. Please review when convenient.

**Pull request merged**